### PR TITLE
Fix invoke exit code

### PR DESCRIPTION
**Pull request by mthenw**
Based on https://github.com/serverless/serverless/pull/2836

## How can we verify it:

```
sls invoke -f ...
echo $?
```

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Change ready for review message below

***Is this ready for review?:*** YES

### CODE REVIEW

**Comment by mthenw** [comment]
@pmuens I don't think that we need to change https://github.com/serverless/serverless/pull/2942/files#diff-1cca2c8b69cb25203eba9dede2461e7dR8. Not sure but looks like this line is no needed. Is there any situation in which unhandled rejection will occur?

**Comment by eahefnawy** [comment]
@mthenw this is a "just in case" scenario ... I suggest we keep it in case we miss handling any rejection (Bluebird suggests we keep it as well).

**Comment by eahefnawy** [comment]
Also, I remember that we had hard time debugging an issue where error happens but is not really thrown/shown as the rejection not handled. That causes the process to just exist without any feedback what so ever. Remember that one @pmuens ? Adding this line made sure we always get the feedback we need.

**Comment by mthenw** [comment]
@eahefnawy yeah, but how is it possible to have unhandled rejection if they are all handled here https://github.com/serverless/serverless/pull/2942/files#diff-1cca2c8b69cb25203eba9dede2461e7dR20

**Comment by pmuens** [comment]
Reviewing this now

File: lib/plugins/aws/invoke/index.js [hunk 1]
```diff
@@ -119,6 +118,10 @@ class AwsInvoke {
       logResult.split('\n').forEach(line => this.consoleLog(formatLambdaLogEvent(line)));
     }
 
+    if (invocationReply.FunctionError) {
```
Comment (pmuens) [inline lib/plugins/aws/invoke/index.js]: Do we need to add smth. like this in `invokeLocal` as well?

Don't have deep knowledge about implementation details here...

**Comment by eahefnawy** [comment]
@mthenw yeah good point. This is exactly what I thought when we faced this bug. But it seems is that a Promise rejection is different from a thrown error somehow. For some reason it was getting "caught" ... I guess that's why Bluebird added this event listener.

Don't have a super strong answer on that one, but I say let's just keep it cause I fear we might face another hard to trace bug again.

File: lib/plugins/aws/invoke/index.js (hunk 1 above)
Comment (eahefnawy) [inline lib/plugins/aws/invoke/index.js]: Don't think so ... `InvocationReply` is returned from the AWS API and is irrelevant to invoke local I think

**Comment by pmuens** [comment]
@eahefnawy yes, I remember that. Maybe we should bring that back somehow. Not sure about the `unhandledRejection` though...

**Pull request merged**